### PR TITLE
chore: Consumer refresh button

### PR DESCRIPTION
**Pull request by Kells512**
**What this PR does / why we need it**:

> A refresh button that can be used for Consumer Groups and Topics

**Verification steps** 

> **Hoover**
> < PROOFOFCONCEPTS > < RefreshButton > < Example1 >
>Screenshot from 2022-09-19 13-35-54

> **Text next to button**
> < PROOFOFCONCEPTS > < RefreshButton > < Example2 >
> Screenshot from 2022-09-19 13-36-10

> **Text to far right*
> < PROOFOFCONCEPTS > < RefreshButton > < Example3  >
> Screenshot from 2022-09-19 13-36-20

### CODE REVIEW

**Comment by Kells512** [comment]
@riccardo-forina I wasn't able to get the test to run on the old pr, but when I opened a new one (this one) it worked

File: src/Kafka/KafkaMessageBrowser/KafkaMessageBrowser.tsx [hunk 1]
```diff
@@ -266,6 +266,11 @@ export const KafkaMessageBrowserConnected: VoidFunctionComponent<
                     </ToolbarItem>
                     <ToolbarItem>
                       <RefreshButton
+                        tooltip={
```
Comment (riccardo-forina) [inline src/Kafka/KafkaMessageBrowser/KafkaMessageBrowser.tsx:269]: I wonder if this shouldn't be the default behavior of the tooltip. So users of the component can simply use it without having to provide the common strings, while users that require some variant of the text can still provide it.

**Review by riccardo-forina — APPROVED** [review]
LGTM, left a comment for something I didn't notice earlier

**Review by Kells512 — COMMENTED** [review]
@riccardo-forina Would you mean having the tooltip altogether being optional to include?

File: src/shared/RefreshButton/RefreshButton.tsx [hunk 2]
```diff
@@ -13,23 +14,31 @@ export const RefreshButton: VoidFunctionComponent<RefreshButtonProps> = ({
...
-      onClick={isDisabled === true ? undefined : onClick}
+    <Tooltip
+      content={
+        tooltip
+          ? tooltip
+          : isRefreshing
+          ? t("refreshing_tooltip")
+          : t("refresh_description")
+      }
     >
```
Comment (Kells512) [inline src/shared/RefreshButton/RefreshButton.tsx:31]: @riccardo-forina does this look alright?

File: src/shared/RefreshButton/RefreshButton.tsx [hunk 3]
```diff
@@ -13,23 +14,31 @@ export const RefreshButton: VoidFunctionComponent<RefreshButtonProps> = ({
   ariaLabel,
   onClick,
   isDisabled,
+  tooltip,
   isRefreshing,
 }) => {
   const { t } = useTranslation("common");
+
```
Comment (riccardo-forina) [inline src/shared/RefreshButton/RefreshButton.tsx:21]:
```suggestion
const defaultTooltip = isRefreshing ? t("refreshing_tooltip") : t("refresh_description");
```

File: src/shared/RefreshButton/RefreshButton.tsx [hunk 4]
```diff
@@ -13,23 +14,31 @@ export const RefreshButton: VoidFunctionComponent<RefreshButtonProps> = ({
...
-      isDisabled={isDisabled}
-      onClick={isDisabled === true ? undefined : onClick}
+    <Tooltip
+      content={
+        tooltip
+          ? tooltip
+          : isRefreshing
+          ? t("refreshing_tooltip")
+          : t("refresh_description")
+      }
```
Comment (riccardo-forina) [inline src/shared/RefreshButton/RefreshButton.tsx:30]:
```suggestion
    <Tooltip
      content={tooltip || defaultTooltip}
```

File: src/shared/RefreshButton/RefreshButton.tsx (hunk 2 above)
Comment (riccardo-forina) [reply src/shared/RefreshButton/RefreshButton.tsx:31]: Yes, but try to avoid nested ternary operators. Something like what I suggested is more readable.

Comment (Kells512) [reply src/shared/RefreshButton/RefreshButton.tsx:25]: I added those suggestions :)

**Review by riccardo-forina — APPROVED** [review]

**Pull request merged**